### PR TITLE
Added a date element.

### PR DESCRIPTION
**Pull request by XanthronWriter**
Add a date element to automatically add the current date to the document. The date can also be formatted and will respect the current language.

To implement this feature, the [`chrono`](https://github.com/chronotope/chrono) crate was added, so the formatting options are the same as in the crate ([formatting options](https://docs.rs/chrono/latest/chrono/format/strftime/index.html)).

Normally, I would have created an issue first to request this feature, but since it was a fairly simple addition and I had already completed it, I refrained from creating an issue.

### CODE REVIEW

File: library/Cargo.toml [hunk 1]
```diff
@@ -32,3 +32,4 @@ unicode-math-class = { git = "https://github.com/typst/unicode-math-class" }
 unicode-script = "0.5"
 unicode-segmentation = "1"
 xi-unicode = "0.3"
+chrono = { version = "0.4", default-features = false, features = ["unstable-locales"]}
```
Comment (CGMossa) [inline library/Cargo.toml:35]: this file needs an empty line at the end

```suggestion
chrono = { version = "0.4", default-features = false, features = ["unstable-locales"] }

```

**Comment by XanthronWriter** [comment]
I have written a small test under `tests/typ/text/date.typ`. Should I move it to the `tests/typ/compiler` folder? Otherwise, I am not sure what should be done.

I am using `chrono` because it supports multiple languages with the `unstable-locales` feature, and they are fairly simple to add.

I could change it to use time-fmt, but then it would be more demanding to implement different languages. At least, I did not find a way to format it in different languages with this crate.

If we continue to use `chrono` I can add a few other languages too. Currently only the const languages `ENGLISH` and `GERMAN` are supported.

**Comment by CGMossa** [comment]
Can I make the really minimalist suggestion and say, let us use `time-fmt`, and then hand-roll support for the `ENGLISH` and `GERMAN` format, and then make that the first step. Then extensive language support could follow in later PRs, and thus that may include something like `chrono`.
`chrono` is falling out of fashion as well, and `time` is taken over, so it isn't really that confident of a move anyways.

I think there should be a compiler test.. Honestly, didn't see your text test.
But I could be very wrong on this one 🤷‍♂️ sorry.

**Comment by XanthronWriter** [comment]
@CGMossa I have changed the crate from `chrono` to `time` and `time-fmt` but since there is no nativ language support, I had to add a little hack by replacing some specifier (`%a`, `%A`, `%b`, `%B`) before the `time-fmt` formatting.

@laurmaedje I agree that a date type would be a better alternativ. But at least we have a starting point right now.

File: library/Cargo.toml [hunk 2]
```diff
@@ -32,3 +32,5 @@ unicode-math-class = { git = "https://github.com/typst/unicode-math-class" }
 unicode-script = "0.5"
 unicode-segmentation = "1"
 xi-unicode = "0.3"
+time = "0.3.20"
+time-fmt = "0.3.8"
```
Comment (CGMossa) [inline library/Cargo.toml:36]:
```suggestion
time-fmt = "0.3.8"

```

**Comment by CGMossa** [comment]
Please note that I'm in no way part of the team. I'm just giving my community member review. It helps other projects, but I could be both factually wrong, and my opinions may be opposite of the typst team.

**Comment by laurmaedje** [comment]
I'm closing this as dates should be a data type not an element. There's some work on the data type approach in #435.

**Pull request closed without merging**